### PR TITLE
GH-8:  Add Connectivity API

### PR DESCRIPTION
**Pull request by jamesmontemagno**
**Checklist**

 - [x] API implementation
 - [x] Tests
 - [x] Samples
 - [x] Documentation

### CODE REVIEW

**Comment by Redth** [comment]
Related to #8

**Comment by jamesmontemagno** [comment]
build

File: Caboodle/Connectivity/Connectivity.android.cs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        {
+            get
+            {
+                ValidatePermission();
+                try
+                {
+                    var currentAccess = NetworkAccess.None;
+                    var manager = Platform.ConnectivityManager;
+
+                    if (Build.VERSION.SdkInt >= BuildVersionCodes.Lollipop)
```
Comment (Redth) [inline Caboodle/Connectivity/Connectivity.android.cs]: Use the new platform.hasapi here?

File: Caboodle/Connectivity/Connectivity.android.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            }
+        }
+
+        public static IEnumerable<ConnectionProfile> Profiles
+        {
+            get
+            {
+                ValidatePermission();
+                var manager = Platform.ConnectivityManager;
+                if (Build.VERSION.SdkInt >= BuildVersionCodes.Lollipop)
```
Comment (Redth) [inline Caboodle/Connectivity/Connectivity.android.cs]: Platform.HasApiLevel

File: Caboodle/Connectivity/Connectivity.android.cs [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        }
+
+        static NetworkAccess IsBetterAccess(NetworkAccess currentAccess, NetworkAccess newAccess) =>
+            newAccess > currentAccess ? newAccess : currentAccess;
+
+        public static NetworkAccess NetworkAccess
+        {
+            get
+            {
+                ValidatePermission();
```
Comment (Redth) [inline Caboodle/Connectivity/Connectivity.android.cs]: Are we ok with potentially throwing here?

File: Caboodle/Connectivity/Connectivity.android.cs (hunk 1 above)
Comment (jamesmontemagno) [reply Caboodle/Connectivity/Connectivity.android.cs]: Updated!

**Pull request merged**